db/consistency_level.cc

```diff
@@ -23,8 +23,37 @@
 #include "locator/abstract_replication_strategy.hh"
 #include "locator/network_topology_strategy.hh"
 #include "heat_load_balance.hh"
+#include "tools/build_info.hh"
 
 namespace db {
+namespace {
+
+// Check the replicas data centers replication factor.
+// If we have nodes with RF=0, the effective replication map is inconsistent.
+void validate_replica_dc_replication_factor(const locator::effective_replication_map& erm, const inet_address_vector_replica_set& replicas) {
+    if constexpr (tools::build_info::is_release_build()) {
+        return;
+    }
+    const auto& topology = erm.get_topology();
+    const auto& replication_strategy = erm.get_replication_strategy();
+    // Applies only to Network Topology Replication Strategy.
+    if (replication_strategy.get_type() != locator::replication_strategy_type::network_topology) {
+        return;
```

Comment:
Cleaner to call into a virtual method of abstract_replication_strategy, no?

Reply:
Which one do you mean? 
`get_replication_factor` in `abstract_replication_strategy` accepts  `const token_metadata&`, but I want to check the data center.

Reply:
Add a new one. Say abstract_replication_strategy::sanity_check_data_center(). The other implementations can {} it.

If we have an interface, why not use it?

Reply:
Created a virtual `sanity_check_read_replicas`; the logic is moved to the overridden version for `network_topology_strategy`.